portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.test.js

```diff
@@ -161,5 +162,49 @@ describe('controllers/utilisation-report-service/utilisation-report-upload', ()
       // Assert
       expect(res._getRedirectUrl()).toEqual('/utilisation-report-upload/confirm-and-send');
     });
+
+    it('filters down the object sent to the API for validation', async () => {
+      // Arrange
+      const { res, req } = httpMocks.createMocks({
+        session: { userToken: 'user-token', user: MOCK_PORTAL_SESSION_USER, logInStatus: 'Valid 2FA' },
+        file: { originalname: 'filename' },
+      });
+
+      const reportJsonWithExtraKeys = [
+        {
+          [UTILISATION_REPORT_HEADERS.BASE_CURRENCY]: 'GBP',
+          keyToRemove: 'test value',
+        },
+        {
+          [UTILISATION_REPORT_HEADERS.BASE_CURRENCY]: 'USD',
+          keyToRemove: 'test value',
+        },
+      ];
+
+      const expectedFilteredReportJson = [
+        {
+          [UTILISATION_REPORT_HEADERS.BASE_CURRENCY]: 'GBP',
+        },
+        {
+          [UTILISATION_REPORT_HEADERS.BASE_CURRENCY]: 'USD',
+        },
+      ];
+
+      jest.mocked(getUploadErrors).mockReturnValueOnce(null);
+
+      jest.mocked(extractCsvData).mockReturnValueOnce({
+        csvJson: reportJsonWithExtraKeys,
+        fileBuffer: null,
+        error: false,
+      });
+
+      jest.spyOn(api, 'generateValidationErrorsForUtilisationReportData').mockReturnValueOnce({ csvValidationErrors: [] });
+
+      // Act
+      await postUtilisationReportUpload(req, res);
+
+      // Assert
+      expect(api.generateValidationErrorsForUtilisationReportData).toHaveBeenCalledWith(expectedFilteredReportJson, expect.any(String), expect.any(String));
```

Comment:
This could be:

```ts
expect(api.generateValidationErrorsForUtilisationReportData).toHaveBeenCalledTimes(1);

const expected = filterReportJsonToRelevantKeys(reportJsonWithExtraKeys);

expect(api.generateValidationErrorsForUtilisationReportData).toHaveBeenCalledWith(expected);
```